src/main/kotlin/com/cognifide/gradle/aem/environment/io/ConfigExpander.kt

```diff
@@ -0,0 +1,93 @@
+package com.cognifide.gradle.aem.environment.io
+
+import com.cognifide.gradle.aem.common.AemExtension
+import com.cognifide.gradle.aem.common.AemTask
+import com.cognifide.gradle.aem.common.file.FileOperations
+import com.cognifide.gradle.aem.common.file.resolver.FileResolver
+import com.cognifide.gradle.aem.environment.EnvironmentException
+import java.io.File
+import org.gradle.util.GFileUtils
+
+class ConfigExpander(private val aem: AemExtension) {
+
+    private val projectDir: String = aem.project.projectDir.path
+
+    private val downloadDir = AemTask.temporaryDir(aem.project, TEMPORARY_DIR)
+
+    private val fileResolver = FileResolver(aem, downloadDir).apply { group(GROUP_EXTRA) {} }
+
+    private val dispatcherDistUrl = aem.environmentOptions.dispatcherDistUrl
+
+    init {
+        setupEnvironmentConfiguration()
+        setupDispatcherDistribution()
+    }
+
+    private fun setupDispatcherDistribution() {
+        val dispatcherModuleFile = dispatcherModuleFile(projectDir)
+        if (dispatcherModuleFile.exists()) {
+            return
+        }
+        if (dispatcherDistUrl.isBlank()) {
+            throw EnvironmentException("Dispatcher distribution URL needs to be configured in order to turn on the environment:" +
+                    "\naem.environment.dispatcher.distUrl")
+        }
+
+        val dispatcherDist = fileResolver.run { dispatcherDistUrl.run { url(this) } }.file
+        val dispatcherDistTree = aem.project.tarTree(dispatcherDist)
+        val sourceDispatcherModuleFile = dispatcherDistTree.find { file -> file.name.startsWith("dispatcher-apache2.4") and file.name.endsWith(".so") }
```

Comment:
version contraint hardcoded; this cannot be done at all; maybe better introduce some task/config parameter holding wildcard expr `Patterns.wildcard(file.name, "dispaycher-apache*")` / I am doing this some many times in GAP; maybe align